src/mono/mono/mini/aot-compiler.c

```diff
@@ -5265,6 +5274,20 @@ MONO_RESTORE_WARNING
 		}
 	}
 
+	if (acfg->aot_opts.export_symbols_outfile && export_symbols->len) {
+		char *export_symbols_out = g_string_free (export_symbols, FALSE);
+		acfg->export_symbols_outfile = fopen (acfg->aot_opts.export_symbols_outfile, "w");
+		g_free (acfg->aot_opts.export_symbols_outfile);
+		if (!acfg->export_symbols_outfile) {
+			fprintf (stderr, "Unable to open specified export_symbols_outfile '%s' to append symbols '%s': %s\n", acfg->aot_opts.export_symbols_outfile, export_symbols_out, strerror (errno));
```

Comment:
cfg->aot_opts.export_symbols_outfile use after free, but as commented above, I don't think you should have freed it at all.